docs/cvl/expr.md

```diff
@@ -550,3 +550,116 @@ As with direct storage access in general, direct storage havoc is experimental a
 * user-defined types such as structs, or arrays/mappings of such types
 * enums
 ```
+
+## Built-in Functions
+
+### Hashing
+
+Many contracts employ hashing in order to create `mapping` keys from more complex data structures.
+A native hashing function in Solidity is `keccak256`.
+We expose `keccak256` in CVL, in order to be able to create such mapping keys directly in CVL.
+The alternative to using the CVL built-in for `keccak256` is to use pre-existing
+functionality from the contract, or creating a dedicated harness function,
+which is usually undesirable from both usability and Prover performance perspectives.
```

Comment:
this seems weirdly specific. I don't know we need to belabor why we add it. Simply describe the primitive, and **then** give an example of how it can be used. As is, it sounds like `keccak256` can or should only be used for mapping keys.

Reply:
right, but in a way it's not exactly the same primitive as in solidity the way ecrecover is virtually the same